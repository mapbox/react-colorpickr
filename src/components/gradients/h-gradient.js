'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import themeable from 'react-themeable';
import { autokey } from '../../autokey';

class HGradient extends React.PureComponent {
  static propTypes = {
    theme: PropTypes.object.isRequired,
    active: PropTypes.bool.isRequired,
    hueBackground: PropTypes.string.isRequired
  };

  render() {
    const theme = autokey(themeable(this.props.theme));
    const { active, hueBackground } = this.props;
    if (!active) return <noscript />;
    return (
      <div>
        <div {...theme('gradient')} style={{ backgroundColor: hueBackground }} />
        <div {...theme('gradient', 'gradientHue')} />
      </div>
    );
  }
}

export default HGradient;
