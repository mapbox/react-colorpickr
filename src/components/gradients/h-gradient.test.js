import React from 'react';
import { render } from '@testing-library/react';
import { HGradient } from './h-gradient';

describe('HGradient', () => {
  test('renders inactive', () => {
    const props = {
      active: false,
      theme: {},
      hueBackground: 'blue'
    };

    const { baseElement } = render(<HGradient {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  test('renders active', () => {
    const props = {
      active: true,
      theme: {},
      hueBackground: 'blue'
    };

    const { baseElement } = render(<HGradient {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
