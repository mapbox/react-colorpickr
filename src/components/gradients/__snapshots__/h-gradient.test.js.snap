// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HGradient renders active 1`] = `
<body>
  <div>
    <div
      style="background-color: blue;"
    />
    <div />
  </div>
</body>
`;

exports[`HGradient renders inactive 1`] = `
<body>
  <div>
    <noscript />
  </div>
</body>
`;
